httpclient/request_id_round_tripper_test.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright © 2024 Acronis International GmbH.
+
+Released under MIT license.
+*/
+
+package httpclient
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/acronis/go-appkit/httpserver/middleware"
+)
+
+func TestNewRequestIDRoundTripper(t *testing.T) {
+	requestID := "12345"
+
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		require.Equal(t, requestID, r.Header.Get("X-Request-ID"))
```

Comment:
All methods from the require package internally call `t.FailNow()`. According to the [documentation](https://pkg.go.dev/testing#T.FailNow):

> FailNow must be called from the goroutine running the test or benchmark function, not from other goroutines created during the test. Calling FailNow does not stop those other goroutines.

To avoid unexpected behavior, please do not use `require` methods from non-main goroutines.